File: data_worker/src/iface/communication/traffic_station.py
from abc import ABC, abstractmethod
from dataclasses import dataclass, field

from src.iface.communication.routing_contract import FlowRouteInput, RouteProfile


@dataclass(slots=True, kw_only=True)
class InboundTrafficRequest:
    flow: FlowRouteInput
    headers: dict[str, str] = field(default_factory=dict)
    payload: str = ""


@dataclass(slots=True, kw_only=True)
class OutboundTrafficRequest:
    flow: FlowRouteInput
    headers: dict[str, str] = field(default_factory=dict)
    payload: str = ""


@dataclass(slots=True, kw_only=True)
class TrafficDecision:
    accepted: bool
    reason: str = ""
    profile: RouteProfile | None = None
    metadata: dict[str, str] = field(default_factory=dict)


@dataclass(slots=True, kw_only=True)
class TrafficDispatchResult:
    profile: RouteProfile | None = None
    target_endpoint: str = ""
    payload: str = ""
    metadata: dict[str, str] = field(default_factory=dict)


class ITrafficStation(ABC):
    """data_worker 的统一流量站点。

    下游接口调用：
      - communication.IRoutingPayloadPipeline.resolve_route_profile / build_outbound_payload
      - authcontrol.IAuthControl.enforce_inbound / prepare_outbound
      - orchestration.IWorkerOrchestrator.handle_task / handle_inbound_rpc
    """

    @abstractmethod
    async def handle_inbound(self, req: InboundTrafficRequest) -> TrafficDecision:
        raise NotImplementedError

    @abstractmethod
    async def send_outbound(self, req: OutboundTrafficRequest) -> TrafficDispatchResult:
        raise NotImplementedError

File: gateway/src/iface/communication/traffic_station.go
package communication

import (
	"context"

	commonif "gateway/src/iface/common"
)

// InboundTrafficRequest 是发送到流量站点的统一入站信封。
type InboundTrafficRequest struct {
	Flow    *commonif.FlowRouteInput
	Headers map[string]string
	Payload string
}

// OutboundTrafficRequest 是从流量站点发出的统一出站信封。
type OutboundTrafficRequest struct {
	Flow    *commonif.FlowRouteInput
	Headers map[string]string
	Payload string
}

// TrafficDecision 是站点的决策输出。
type TrafficDecision struct {
	Accepted bool
	Reason   string

	Profile  *commonif.RouteProfile
	Metadata map[string]string
}

// TrafficDispatchResult 是站点的出站派发输出。
type TrafficDispatchResult struct {
	Profile        *commonif.RouteProfile
	TargetEndpoint string
	Payload        string
	Metadata       map[string]string
}

// ITrafficStation 定义模块级统一流量站点契约。
//
// 下游接口调用：
//   - communication.IRoutingPayloadPipeline.ResolveRouteProfile / BuildOutboundPayload
//   - authcontrol.IGatewayAuthControl.Enforce
//   - orchestration.IForwardingOrchestrator.HandleBusinessForward / HandleExternalAuthForward
type ITrafficStation interface {
	HandleInbound(ctx context.Context, req *InboundTrafficRequest) (*TrafficDecision, error)
	SendOutbound(ctx context.Context, req *OutboundTrafficRequest) (*TrafficDispatchResult, error)
}
